### ISSUE TITLE
logistic regression - intercept Odds Ratio wrong

### ISSUE THREAD

**Issue by DesmondCampbell** [chrchang/plink-ng#102]
Hi Chris,

When I run "plink --logistic intercept"
 I get a negative number for OR output (isn't it supposed to be the Odds Ratio?) for the intercept term.
ORs can't be negative so clearly wrong.

When I run "plink --logistic intercept beta"
 I get the same negative number for the BETA output for the intercept term

This also applies to the SE output.

Also the L95 and U95 outputs (--ci flag) appear approximately correct for the BETA estimate
but up the spout for the OR estimate.

The code fragment below demonstrates both above regressions.
Compare the two INTERCEPT lines of output.

regards
Desmond
( of Hong Kong)

```
$ plink --bfile ../../data/ukb_17333_bmi_locke/ukb_17333_bmi_locke --snp rs977747 --ci 0.95 --pheno ukb_17333_bmi_locke.pheno --pheno-name iRetiredNotEmp --covar ukb_17333_bmi_locke.pheno --covar-name f.21003.0.0,PGC1 --logistic intercept --out assoc_iRetiredNotEmp_ageSexCentreGpc
PLINK v1.90b6.6 64-bit (10 Oct 2018)           www.cog-genomics.org/plink/1.9/
(C) 2005-2018 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to assoc_iRetiredNotEmp_ageSexCentreGpc.log.
Options in effect:
  --bfile ../../data/ukb_17333_bmi_locke/ukb_17333_bmi_locke
  --ci 0.95
  --covar ukb_17333_bmi_locke.pheno
  --covar-name f.21003.0.0,PGC1
  --logistic intercept
  --out assoc_iRetiredNotEmp_ageSexCentreGpc
  --pheno ukb_17333_bmi_locke.pheno
  --pheno-name iRetiredNotEmp
  --snp rs977747

8078 MB RAM detected; reserving 4039 MB for main workspace.
Allocated 3029 MB successfully, after larger attempt(s) failed.
1 out of 96 variants loaded from .bim file.
487395 people (223033 males, 264362 females) loaded from .fam.
441138 phenotype values present after --pheno.
Using 1 thread (no multithreaded calculations invoked).
--covar: 2 out of 165 covariates loaded.
17 people had missing value(s).
Before main variant filters, 487395 founders and 0 nonfounders present.
Calculating allele frequencies... done.
Total genotyping rate is 0.994425.
1 variant and 487395 people pass filters and QC.
Among remaining phenotypes, 162319 are cases and 278819 are controls.  (46257
phenotypes are missing.)
Writing logistic model association results to
assoc_iRetiredNotEmp_ageSexCentreGpc.assoc.logistic ... done.

dc267s@DELL-D1MWV22 /cygdrive/z/Projects/Biobank Work Study S00101/Desmond/analyses/bmi_mr_9jan19
$ cat assoc_iRetiredNotEmp_ageSexCentreGpc.assoc.logistic
 CHR        SNP         BP   A1       TEST    NMISS         OR       SE      L95      U95         STAT            P
   1   rs977747   47684677    T        ADD   438781      1.013 0.006418    1.001    1.026        2.075      0.03801
   1   rs977747   47684677    T f.21003.0.0   438781      1.425 0.001111    1.422    1.428        318.8            0
   1   rs977747   47684677    T       PGC1   438781      0.998 0.0001197   0.9978   0.9983       -16.47    6.179e-61
   1   rs977747   47684677    T  INTERCEPT   438781     -21.65  0.06823 3.463e-10 4.525e-10           NA           NA

dc267s@DELL-D1MWV22 /cygdrive/z/Projects/Biobank Work Study S00101/Desmond/analyses/bmi_mr_9jan19
$ plink --bfile ../../data/ukb_17333_bmi_locke/ukb_17333_bmi_locke --snp rs977747 --ci 0.95 --pheno ukb_17333_bmi_locke.pheno --pheno-name iRetiredNotEmp --covar ukb_17333_bmi_locke.pheno --covar-name f.21003.0.0,PGC1 --logistic intercept beta --out assoc_iRetiredNotEmp_ageSexCentreGpc
PLINK v1.90b6.6 64-bit (10 Oct 2018)           www.cog-genomics.org/plink/1.9/
(C) 2005-2018 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to assoc_iRetiredNotEmp_ageSexCentreGpc.log.
Options in effect:
  --bfile ../../data/ukb_17333_bmi_locke/ukb_17333_bmi_locke
  --ci 0.95
  --covar ukb_17333_bmi_locke.pheno
  --covar-name f.21003.0.0,PGC1
  --logistic intercept beta
  --out assoc_iRetiredNotEmp_ageSexCentreGpc
  --pheno ukb_17333_bmi_locke.pheno
  --pheno-name iRetiredNotEmp
  --snp rs977747

8078 MB RAM detected; reserving 4039 MB for main workspace.
Allocated 3029 MB successfully, after larger attempt(s) failed.
1 out of 96 variants loaded from .bim file.
487395 people (223033 males, 264362 females) loaded from .fam.
441138 phenotype values present after --pheno.
Using 1 thread (no multithreaded calculations invoked).
--covar: 2 out of 165 covariates loaded.
17 people had missing value(s).
Before main variant filters, 487395 founders and 0 nonfounders present.
Calculating allele frequencies... done.
Total genotyping rate is 0.994425.
1 variant and 487395 people pass filters and QC.
Among remaining phenotypes, 162319 are cases and 278819 are controls.  (46257
phenotypes are missing.)
Writing logistic model association results to
assoc_iRetiredNotEmp_ageSexCentreGpc.assoc.logistic ... done.

dc267s@DELL-D1MWV22 /cygdrive/z/Projects/Biobank Work Study S00101/Desmond/analyses/bmi_mr_9jan19
$ cat assoc_iRetiredNotEmp_ageSexCentreGpc.assoc.logistic
 CHR        SNP         BP   A1       TEST    NMISS       BETA       SE      L95      U95         STAT            P
   1   rs977747   47684677    T        ADD   438781    0.01332 0.006418 0.0007366   0.0259        2.075      0.03801
   1   rs977747   47684677    T f.21003.0.0   438781     0.3541 0.001111   0.3519   0.3562        318.8            0
   1   rs977747   47684677    T       PGC1   438781  -0.001972 0.0001197 -0.002206 -0.001737       -16.47    6.179e-61
   1   rs977747   47684677    T  INTERCEPT   438781     -21.65  0.06823   -21.78   -21.52           NA           NA
```

**Comment by chrchang** [comment]
Oops, the intercept-reporting code didn't check for odds-ratio output.  Will post a fix later today.

**Comment by chrchang** [comment]
Should be fixed by commit 8beb467.

**Comment by DesmondCampbell** [comment]
Hi Chris,

Just downloaded and tried this for PLINK v1.90b6.9 64-bit (4 Mar 2019).
Still some problems.

In the *.assoc.logistic file the standard error reported seems to be the same regardless of whether you report Odds Ratio or Reg Coeff (See SE column in the tables below) .
The reported SE seems to be the one for Reg Coeff (See the is_It_1.96 column in the tables below).
This problem isn't just for INTERCEPT but also for the other regressors.

The L95 and U95 columns look right but I haven't checked.

Below I give plink invocations and the results they produce for specimen
--logistic intercept
--logistic beta intercept

Yours in pedantry,

Desmond


plink --bfile ../../data/ukb_17333_bmi_locke/ukb_17333_bmi_locke --snp rs977747 --ci 0.95 --pheno ../qc/ukb_17333_bmi_locke.pheno --pheno-name iRetiredNotEmp --covar ../qc/ukb_17333_bmi_locke.pheno --covar-name f.21003.0.0,PGC1 --logistic intercept
PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/
(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to plink.log.
Options in effect:
  --bfile ../../data/ukb_17333_bmi_locke/ukb_17333_bmi_locke
  --ci 0.95
  --covar ../qc/ukb_17333_bmi_locke.pheno
  --covar-name f.21003.0.0,PGC1
  --logistic intercept
  --pheno ../qc/ukb_17333_bmi_locke.pheno
  --pheno-name iRetiredNotEmp
  --snp rs977747
...
Writing logistic model association results to plink.assoc.logistic ... done.

Gives the following results

CHR | SNP | BP | A1 | TEST | NMISS | OR | SE | L95 | U95 | STAT | P | 95pcCI | nofSEs | is_It_1.96
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
1 | rs977747 | 47684677 | T | ADD | 438781 | 1.013 | 0.006418 | 1.001 | 1.026 | 2.075 | 0.03801 | 0.025 | 3.895294 | 1.947647
1 | rs977747 | 47684677 | T | f.21003.0.0 | 438781 | 1.425 | 0.001111 | 1.422 | 1.428 | 318.8 | 0 | 0.006 | 5.40054 | 2.70027
1 | rs977747 | 47684677 | T | PGC1 | 438781 | 0.998 | 0.00012 | 0.9978 | 0.9983 | -16.47 | 6.18E-61 | 0.0005 | 4.177109 | 2.088555
1 | rs977747 | 47684677 | T | INTERCEPT | 438781 | 3.96E-10 | 0.06823 | 3.46E-10 | 4.53E-10 | NA | NA | 1.06E-10 | 1.56E-09 | 7.78E-10

Whereas
plink --bfile ../../data/ukb_17333_bmi_locke/ukb_17333_bmi_locke --snp rs977747 --ci 0.95 --pheno ../qc/ukb_17333_bmi_locke.pheno --pheno-name iRetiredNotEmp --covar ../qc/ukb_17333_bmi_locke.pheno --covar-name f.21003.0.0,PGC1 --logistic beta intercept
PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/
(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to plink.log.
Options in effect:
  --bfile ../../data/ukb_17333_bmi_locke/ukb_17333_bmi_locke
  --ci 0.95
  --covar ../qc/ukb_17333_bmi_locke.pheno
  --covar-name f.21003.0.0,PGC1
  --logistic beta intercept
  --pheno ../qc/ukb_17333_bmi_locke.pheno
  --pheno-name iRetiredNotEmp
  --snp rs977747
...
Writing logistic model association results to plink.assoc.logistic ... done.

gives the following results


CHR | SNP | BP | A1 | TEST | NMISS | BETA | SE | L95 | U95 | STAT | P | 95pcCI | nofSEs | is_It_1.96
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
1 | rs977747 | 47684677 | T | ADD | 438781 | 0.01332 | 0.006418 | 0.000737 | 0.0259 | 2.075 | 0.03801 | 0.025163 | 3.920754 | 1.960377
1 | rs977747 | 47684677 | T | f.21003.0.0 | 438781 | 0.3541 | 0.001111 | 0.3519 | 0.3562 | 318.8 | 0 | 0.0043 | 3.870387 | 1.935194
1 | rs977747 | 47684677 | T | PGC1 | 438781 | -0.00197 | 0.00012 | -0.00221 | -0.00174 | -16.47 | 6.18E-61 | 0.000469 | 3.918129 | 1.959064
1 | rs977747 | 47684677 | T | INTERCEPT | 438781 | -21.65 | 0.06823 | -21.78 | -21.52 | NA | NA | 0.26 | 3.81064 | 1.90532

**Comment by chrchang** [comment]
This is intentional and documented.  (With that said, I'll consider renaming the "SE" column to "BETA_SE" when this situation comes up in plink 2.0, since yes, it's not obvious.)

**Comment by DesmondCampbell** [comment]
OK, well feel free to close the issue if your happy.

**Comment by chrchang** [comment]
Addressed this with commit fdb2636 (column name is now "LOG(OR)_SE").
